rust/worker/src/execution/operators/group.rs

```diff
@@ -0,0 +1,223 @@
+use crate::errors::{ChromaError, ErrorCodes};
+use crate::execution::data::data_chunk::DataChunk;
+use crate::execution::operator::Operator;
+use async_trait::async_trait;
+use std::collections::HashMap;
+use thiserror::Error;
+
+#[derive(Debug)]
+pub struct GroupOperator {
+    // Approximatedly the number of records in a group. Since we are trying to
+    // group the records by id, which can casue the group size to be larger than this
+    // value.
+    pub(crate) group_size: usize,
+}
+
+/// The input to the group operator.
+/// # Parameters
+/// * `records` - The records to group.
+#[derive(Debug)]
+pub struct GroupInput {
+    pub(crate) records: DataChunk,
+}
+
+impl GroupInput {
+    /// Create a new group input.
+    /// # Parameters
+    /// * `records` - The records to group.
+    pub fn new(records: DataChunk) -> Self {
+        GroupInput { records }
+    }
+}
+
+/// The output of the group operator.
+/// # Parameters
+/// * `records` - The grouped records.
+#[derive(Debug)]
+pub struct GroupOutput {
+    pub(crate) records: Vec<DataChunk>,
+}
+
+#[derive(Debug, Error)]
+pub enum GroupError {
+    #[error("Failed to group records.")]
+    GroupError,
+}
+
+impl ChromaError for GroupError {
+    fn code(&self) -> ErrorCodes {
+        match self {
+            GroupError::GroupError => ErrorCodes::Internal,
+        }
+    }
+}
+
+pub type GroupResult = Result<GroupOutput, GroupError>;
+
+impl GroupOperator {
+    pub fn new(group_size: usize) -> Box<Self> {
+        Box::new(GroupOperator { group_size })
+    }
+
+    pub fn group_by_id(&self, records: &DataChunk) -> Vec<DataChunk> {
```

Comment:
Do we think its odd that we group_by and return many data chunks as opposed to just presenting a reordered data structure?